### PR TITLE
Improve type safety according to pyright

### PR DESCRIPTION
**Pull request by tristan957**
Pyright found many issues that mypy doesn't seem to want to catch or mypy isn't configured to catch.

### CODE REVIEW

File: test_runner/regress/test_wal_acceptor.py [hunk 1]
```diff
@@ -1506,15 +1506,10 @@ def start_safekeeper(self, i):
...
             auth_token=None,
         )
-        try:
-            safekeeper_process = start_in_background(
-                cmd, safekeeper_dir, "safekeeper.log", safekeeper_client.check_status
-            )
-            return safekeeper_process
-        except Exception as e:
-            log.error(e)
-            safekeeper_process.kill()
```
[the remark this answers is not in the document]
Comment (bayandin) [reply test_runner/regress/test_wal_acceptor.py:1516]: It seems `safekeeper_process` is unbound, so `except` block is basically a no-op

[the remark this answers is not in the document]
Comment (tristan957) [reply test_runner/regress/test_wal_acceptor.py:1516]: You'll never have a `safekeeper_process` to kill. If you end up in the exception, you've thrown trying to create the process. The specific error here was that the variable is possibly unbound.

Is there something that I'm missing in regard to this snippet?

[the remark this answers is not in the document]
Comment (jcsp) [reply test_runner/regress/test_wal_acceptor.py:1516]: Yep, fair enough.

[the remark this answers is not in the document]
Comment (tristan957) [reply test_runner/regress/test_wal_acceptor.py:1516]: Thanks for the review!

File: test_runner/fixtures/benchmark_fixture.py [hunk 2]
```diff
@@ -80,7 +80,13 @@ def parse_from_stdout(
     ):
         stdout_lines = stdout.splitlines()
 
+        number_of_clients = 0
+        number_of_threads = 0
+        number_of_transactions_actually_processed = 0
+        latency_average: float = 0
         latency_stddev = None
+        tps: float = 0
```
Comment (bayandin) [inline test_runner/fixtures/benchmark_fixture.py:88]: Optionally, if this works, maybe for consistensy: 
```suggestion
        tps = 0.0
```

**Review by bayandin — APPROVED** [review]
Thanks Tristan!

File: test_runner/fixtures/benchmark_fixture.py (hunk 2 above)
Comment (tristan957) [inline test_runner/fixtures/benchmark_fixture.py:88]:
```suggestion
        latency_average = 0.0
        latency_stddev = None
        tps = 0.0
```

**Pull request merged**